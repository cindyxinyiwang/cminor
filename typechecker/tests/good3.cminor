// statements

m: boolean;
b: function void () = {

	// declaration
	
	x: integer = 20;
	// if statement

	if (x > 0) {
		m   = true;
	} else {
		x = x+1;
		x++;
	}
	// if statement

	if (x*3 <= 12) {
		print "this is awesome", x, 12;
	}
	
	// for
	
	for (x =0; x < 12; x++) {
		m = !m;
	}
	for (;x < 12;x++) {
		m = !!m;
	}
	


	{
		x = x*1;
		return;
	}
	
}