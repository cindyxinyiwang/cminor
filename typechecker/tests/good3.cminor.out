x resovles to local 0
m resovles to global m
x resovles to local 0
x resovles to local 0
x resovles to local 0
x resovles to local 0
x resovles to local 0
x resovles to local 0
x resovles to local 0
x resovles to local 0
m resovles to global m
m resovles to global m
x resovles to local 0
x resovles to local 0
m resovles to global m
m resovles to global m
x resovles to local 0
x resovles to local 0
